src/pages/patientView/pathwayMapper/PatientViewPathwayMapper.tsx

```diff
@@ -0,0 +1,291 @@
+import * as React from 'react';
+import _ from 'lodash';
+import { PatientViewPageStore } from '../clinicalInformation/PatientViewPageStore';
+import 'pathway-mapper/dist/base.css';
+import PathwayMapperTable, {
+    IPathwayMapperTable,
+    IPathwayMapperTableColumnType,
+} from '../../../shared/lib/pathwayMapper/PathwayMapperTable';
+import { observer } from 'mobx-react';
+import autobind from 'autobind-decorator';
+import { observable, computed, action } from 'mobx';
+import { Row } from 'react-bootstrap';
+
+import { AppStore } from 'AppStore';
+import { remoteData } from 'cbioportal-frontend-commons';
+import { fetchGenes } from 'shared/lib/StoreUtils';
+import 'react-toastify/dist/ReactToastify.css';
+import { ToastContainer } from 'react-toastify';
+import PatientViewUrlWrapper from '../PatientViewUrlWrapper';
+import { getGeneticTrackRuleSetParams } from 'shared/components/oncoprint/OncoprintUtils';
+import PathwayMapper, { ICBioData } from 'pathway-mapper';
+
+import 'cytoscape-panzoom/cytoscape.js-panzoom.css';
+import 'cytoscape-navigator/cytoscape.js-navigator.css';
+import 'react-toastify/dist/ReactToastify.css';
+
+const alterationFrequencyData: ICBioData[] = [];
+
+interface IPatientViewPathwayMapperProps {
+    store: PatientViewPageStore;
+    appStore: AppStore;
+    urlWrapper: PatientViewUrlWrapper;
+}
+
+const DEFAULT_RULESET_PARAMS = getGeneticTrackRuleSetParams(true, true, true);
+
+@observer
+export default class PatientViewPathwayMapper extends React.Component<
+    IPatientViewPathwayMapperProps
+> {
+    private accumulatedAlterationFrequencyDataForNonQueryGenes: ICBioData[];
+    private readonly accumulatedValidGenes: { [gene: string]: boolean };
+
+    @observable
+    private selectedPathway = '';
+
+    @observable
+    private newGenesFromPathway: string[];
+
+    private readonly validNonQueryGenes = remoteData<string[]>({
+        invoke: async () => {
+            const genes = await fetchGenes(this.newGenesFromPathway);
+            return genes.map(gene => gene.hugoGeneSymbol);
+        },
+        onResult: (genes: string[]) => {},
+    });
```

Comment:
We do not need this for patient view page, since we don't really query patients with gene.